src/Microsoft.AspNet.OData.Shared/Query/Expressions/SelectExpandWrapperJsonConverter.cs

```diff
@@ -0,0 +1,311 @@
+
+        /// <inheritdoc/>
+        public override bool CanConvert(Type typeToConvert)
+        {
+            if (typeToConvert == null)
+            {
+                throw Error.ArgumentNull(nameof(typeToConvert));
+            }
+
+            if (!typeToConvert.IsGenericType)
+            {
+                return false;
+            }
+
+            return typeof(ISelectExpandWrapper).IsAssignableFrom(typeToConvert);
+        }
+
+        /// <inheritdoc/>
+        public override JsonConverter CreateConverter(
+            Type typeToConvert,
+            JsonSerializerOptions options)
+        {
+            if (typeToConvert == null || !typeToConvert.IsGenericType)
+            {
+                return null;
+            }
+
+            Type genericType = typeToConvert.GetGenericTypeDefinition();
+            Type type = typeToConvert.GetGenericArguments()[0];
+
+            if (genericType == typeof(SelectExpandBinder.SelectSome<>))
+            {
+                return (JsonConverter)Activator.CreateInstance(
+                    typeof(SelectSomeOfTJsonConverter<>).MakeGenericType(new Type[] { type }));
+            }
+
+            if (genericType == typeof(SelectExpandBinder.SelectSomeAndInheritance<>))
+            {
+                return (JsonConverter)Activator.CreateInstance(
+                    typeof(SelectSomeAndInheritanceOfTJsonConverter<>).MakeGenericType(new Type[] { type }));
+            }
+
+            if (genericType == typeof(SelectExpandBinder.SelectAll<>))
+            {
+                return (JsonConverter)Activator.CreateInstance(
+                    typeof(SelectAllOfTJsonConverter<>).MakeGenericType(new Type[] { type }));
+            }
+
+            if (genericType == typeof(SelectExpandBinder.SelectAllAndExpand<>))
+            {
+                return (JsonConverter)Activator.CreateInstance(
+                    typeof(SelectAllAndExpandOfTJsonConverter<>).MakeGenericType(new Type[] { type }));
+            }
+
+            if (genericType == typeof(SelectExpandWrapper<>))
+            {
+                return (JsonConverter)Activator.CreateInstance(
+                    typeof(SelectExpandWrapperOfTJsonConverter<>).MakeGenericType(new Type[] { type }));
+            }
+
+            return null;
+        }
+
+        private class JsonPropertyNameMapper : IPropertyMapper
+        {
+            private IEdmModel _model;
+            private IEdmStructuredType _type;
+
+            public JsonPropertyNameMapper(IEdmModel model, IEdmStructuredType type)
+            {
+                _model = model;
+                _type = type;
+            }
+
+            public string MapProperty(string propertyName)
+            {
+                IEdmProperty property = _type.Properties().First(s => s.Name == propertyName);
+                PropertyInfo propertyInfo = GetPropertyInfo(property);
+
+                Newtonsoft.Json.JsonPropertyAttribute jsonPropertyAttribute;
```

Comment:
nit: add Newtonsoft.Json as import

Reply:
@ElizabethOkerio Because there are types named the same between Newtonsoft.Json and System.Text.Json libraries and we only use the Newtonsoft types a few places, I thought it'd be cleaner to not import Newtonsoft.Json. It'd require me to alias the import anyway to prevent types from clashing.